hrabsence/CRM/HRAbsence/BAO/HRAbsenceType.php

```diff
@@ -42,44 +42,68 @@ public static function create($params) {
-          'is_active' => 1,
-          'is_optgroup' => 0,
-          'is_default' => 0,
-          'grouping' => 'Timesheet',
-        );
-        $resultCreateActivityType = civicrm_api3('activity_type', 'create', $paramsCreate);
-        $debitActivityTypeId = $resultCreateActivityType['values'][$resultCreateActivityType["id"]]['value'];
+    if (CRM_Utils_Array::value('allow_debits', $params)) {
+      if (empty($params['debit_activity_type_id'])) {
+        $weight = count($activityTypesResult["values"]);
+        $debitActivityLabel = $params['title'];
+        $debitActivityTypeId = array_search($debitActivityLabel, $activityTypesResult["values"]);
+        if (!$debitActivityTypeId) {
+          $weight = $weight + 1;
+          $paramsCreate = array(
+            'weight' => $weight,
+            'label' => $debitActivityLabel,
+            'filter' => 1,
+            'is_active' => 1,
+            'is_optgroup' => 0,
+            'is_default' => 0,
+            'grouping' => 'Timesheet',
+          );
+          $resultCreateActivityType = civicrm_api3('activity_type', 'create', $paramsCreate);
+          $debitActivityTypeId = $resultCreateActivityType['values'][$resultCreateActivityType["id"]]['value'];
+        }
+        $params["debit_activity_type_id"] = $debitActivityTypeId;
+      } else {
+        $optionValue = civicrm_api3('OptionValue', 'get', array(
+          'sequential' => 1,
+          'option_group_id' => "activity_type",
+          'value' => $params['debit_activity_type_id'],
+        ));
+        civicrm_api3('OptionValue', 'create', array(
+          'id' => $optionValue['id'],
+          'label' => $params['title'],
+        ));
       }
-      $params["debit_activity_type_id"] = $debitActivityTypeId;
     }
-    if (CRM_Utils_Array::value('allow_credits', $params) && empty($params["credit_activity_type_id"])) {
-      $weight = count($activityTypesResult["values"]);
-      $creditActivityLabel = ts('%1 (Credit)', array(1 => $params["title"]));
-      $creditActivityTypeId = array_search($creditActivityLabel, $activityTypesResult["values"]);
-      if (!$creditActivityTypeId) {
-        $weight = $weight + 1;
-        $paramsCreate = array(
-          'weight' => $weight,
-          'label' => $creditActivityLabel,
-          'filter' => 1,
-          'is_active' => 1,
-          'is_optgroup' => 0,
-          'is_default' => 0,
-          'grouping' => 'Timesheet',
-        );
-        $resultCreateActivityType = civicrm_api3('activity_type', 'create', $paramsCreate);
-        $creditActivityTypeId = $resultCreateActivityType['values'][$resultCreateActivityType["id"]]['value'];
-        $params["credit_activity_type_id"] = $creditActivityTypeId;
+    if (CRM_Utils_Array::value('allow_credits', $params)){
+      if (empty($params['credit_activity_type_id'])) {
+        $weight = count($activityTypesResult["values"]);
+        $creditActivityLabel = ts('%1 (Credit)', array(1 => $params["title"]));
```

Comment:
What about moving this line to outside the if/else so you don't need to duplicate it when updating the Activity Type label?